src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -355,7 +357,8 @@ private void OnInfoMessage(object sender, SqlInfoMessageEventArgs args)
                 // Did the database context change (error code 5701)?
                 if (error.Number == DatabaseContextChangeErrorNumber)
                 {
-                    ConnectionService.Instance.ChangeConnectionDatabaseContext(editorConnection.OwnerUri, conn.Database);
+                    newDatabase = conn.Database;
+                    //ConnectionService.Instance.ChangeConnectionDatabaseContext(editorConnection.OwnerUri, conn.Database);
```

Comment:
I honestly don't remember my reasoning for this. I'll investigate and revert if needed.